c_api/netutil_c_api.go

```diff
@@ -20,9 +20,9 @@ struct EnvResponse {
 	struct EnvData *pEnvs;
 };
 
-
-#define NETUTIL_NUM_IPS            10
-#define NETUTIL_NUM_NETWORKSTATUS  10
+#define NETUTIL_NUM_IPS			10
```

Comment:
Question: Long term, are we leaning towards:
 1) the fixed arrays with a big enough number of entries
 2) making the C side query for number of instances then allocate
 3) making the C side allocate the allocate the number based on deployment and maybe provide defaults/suggstions

Reply:
Could you please elaborate a little bit what is a deployment in `3` ?

Reply:
For 3), look at app_sample.c for where it calls GetEnv(). Same logic but NETUTIL_NUM_DEFAULT_ENVS is defined on the GO side. The C side allocates dynamically as is, and can use NETUTIL_NUM_DEFAULT_ENVS, NETUTIL_NUM_DEFAULT_ENVS*2, or any number it chooses.

Reply:
The difference between `GetEnv` and `GetNetworkStatus` is whether the length of array is provided by user or hardcoded in the code. In the former case, user has the control to allocate default/suggested memory size which is more flexible. Both may hit `SIZE_ERROR` if the length of array is not long enough. +1 on this approach.
If user really wants to allocate exact size of memory which API output occupies, then 2 is good for him.
From API design point of view, we can try to provide APIs that have deterministic output length, for example, `GetNetworkInterface`, if use wants to run an application that requires two interfaces, then he can use method `3` to only allocate 2*Interface_struct; but this may have problem when later user wants to get other interfaces, because there is no order of interface guaranteed by `GetNetworkInterface`.